src/karapace/rapu.py

```diff
@@ -202,10 +202,18 @@ def check_rest_headers(self, request: HTTPRequest) -> dict:
         default_content = "application/vnd.kafka.json.v2+json"
         default_accept = "*/*"
         result = {"content_type": default_content}
-        content_matcher = REST_CONTENT_TYPE_RE.search(
-            cgi.parse_header(request.get_header("Content-Type", default_content))[0]
-        )
-        accept_matcher = REST_ACCEPT_RE.search(cgi.parse_header(request.get_header("Accept", default_accept))[0])
+
+        # Handle Content-Type
+        msg = Message()
```

Comment:
Would it make sense to use a single `Message()` class for all the parsing? 
Since we mostly only update the `Content-Type`, i.e. `msg["Content-Type"] = xyz`, but I am unsure how it'd affect the `msg.get_params()` call, ideally should be fine, but not sure.

Reply:
Message class is mutable, and it might cause side effects during concurrent requests. I'd better leave it inside the function.